clippy_lints/src/useless_conversion.rs

```diff
@@ -195,3 +197,12 @@ impl<'tcx> LateLintPass<'tcx> for UselessConversion {
         }
     }
 }
+
+/// `cx.qpath_res` seems to return `AssocFn` so we do this instead
+fn is_ty_alias(qpath: &QPath<'_>) -> bool {
```

Comment:
This could well be in `clippy_utils`, I think.

Reply:
done, though I did have to enable the feature `if_let_guards`.